modules/SofaUserInteraction/RayContact.h

```diff
@@ -82,11 +82,11 @@ class RayContact : public BaseRayContact
 
     void setDetectionOutputs(core::collision::DetectionOutputVector* outputs) override
     {
-        OutputVector* o = static_cast<OutputVector*>(outputs);
+        const core::collision::DetectionOutput* o = outputs->getContacts();
         //collisions = outputs;
-        collisions.resize(o->size());
-        for (unsigned int i=0; i< o->size(); ++i)
-            collisions[i] = &(*o)[i];
+        collisions.resize(outputs->size());
+        for (unsigned int i=0; i< outputs->size(); ++i)
+            collisions[i] = &o[i];
```

Comment:
Convert to `std::copy` ?

Reply:
outputs is the interface to interact with the contacts . 
it implements a size() function but it doesn't inherit anymore from std::vector so it can't provide a begin and end iterators

Reply:
It may be a good time to do so. You can alternatively use `std::begin` and `std::end`